cmd/web.go

```diff
@@ -48,6 +48,26 @@ and it takes care of all the other things for you`,
 	},
 }
 
+func runHTTPRedirector() {
+	source := setting.HTTPAddr + ":80"
```

Comment:
I think that port 80 is a requirement with this feature.

My intention was to address the simple case of someone typing a bare domain name into their web browser location bar, hitting return and ending up at the right place even when gitea is being served on https.

For example a company has an internal server on their intranet with the name: git.example.com.  The actually URL is https://git.example.com - but end users may type git.example.com into their location bar and hit return.  Without the redirect the browser gives them an error, with the redirect they end up at the right URL.